notebooks/ima_viewer/ima_tools.py

```diff
@@ -0,0 +1,747 @@
+    """
+    Helper function to grab the data from a specific
+    extension, find the normalization factors from the
+    percentage limits either set by the user or from the
+    default percentages.
+
+    Parameters
+    ----------
+    ext : int
+        Integer corresponding to the extension of the
+        fits file.
+    fits_file : fits
+        Fits file at hand.
+    pct_limits : tuple of float
+        Percentage limits as set by user. Default is a
+        lower bound of 50% and upper bound of 99.5%.
+
+    Returns
+    -------
+    data : array
+        Array of data from the specified extension.
+    limits : tuple of float
+        Lower and upper limits, specific to the extension.
+    """
+    data = fits_file[ext].data
+    data_min = np.nanpercentile(data, pct_limits[0])
+    data_max = np.nanpercentile(data, pct_limits[1])
+    limits = (data_min, data_max)
+    return data, limits
+
+def get_indices(fits_file, ext_type):
+    """
+    Helper function to grab the indices that are
+    relevant and match the extension type specified.
+
+    Parameters
+    ----------
+    fits_file : fits
+        Fits file at hand.
+    ext_type : str
+        Extension type.
+
+    Returns
+    -------
+    indices : list of int
+        List of extensions, ordered by time.
+    """
+    indices = [i for i, ext in enumerate(fits_file) if ext.name == ext_type]
+    indices.reverse()
+    return indices
+
+def make_interactive_plot(filepath, ext_type='SCI'):
+    """
+    Function to make and display an interactive
+    plot for a WFC3/IR IMA file.
+
+    Parameters
+    ----------
+    filepath : str
+        String representation of the path to the
+        IMA file.
+    ext_type : str
+        Extension of the IMA fits file that should
+        be displayed. Valid inputs are 'SCI', 'ERR',
+        and 'DQ'. Default is 'SCI'.
+
+    Returns
+    -------
+    plot : tuple or None
+        If any input is invalid, then ``plot`` is None,
+        otherwise, ``plot`` is a tuple of 5 items:
+            fig : `matplotlib.figure.Figure`
+            ax :  `matplotlib.axes._subplots.AxesSubplot`
+            w : `ipywidgets.widgets.interaction.interactive`
+                Contains 3 widgets:
+                    cmap :
+                    ext :
+                    lim_slider :
+            filepath : str
+            ext_type : str
+    """
+    inputs = verify_input(filepath, ext_type)
+
+    if inputs == None:
+        plot = None
+
+    else:
+        fits_file = fits.open(inputs[0])
+        indices = get_indices(fits_file, inputs[1])
+        rootname = fits_file[0].header['ROOTNAME']
+        fig, ax = plt.subplots()
+        data, limits = get_data_and_norm_lims(indices[0], fits_file)
+        img = ax.imshow(data, origin='lower',
+                        cmap='viridis_r',
+                        norm=Normalize(vmin=limits[0], vmax=limits[1]))
+        cb = fig.colorbar(mappable=img, label='electrons', ax=ax)
```

Comment:
Are IMAs electrons or electron per second? Also when plotting the DQ array it still has units of electrons